core/src/net/mod.rs

```diff
@@ -54,6 +55,28 @@ impl From<Transport> for Protocol {
     }
 }
 
+/// Session identifier, part of the `NetMessage` struct. Managed by Kompact internally, may be read
+/// by users to detect session loss, indicated by different SessionId's of NetMessages.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct SessionId(Uuid);
+
+impl SessionId {
+    /// Generates a new SessionId
+    pub fn new_unique() -> SessionId {
+        SessionId(Uuid::new_v4())
+    }
+
+    /// For Serialisation of SessionId's
+    pub fn as_u128(&self) -> u128 {
+        self.0.as_u128()
+    }
+
+    /// For Deserialisation of SessionId's
+    pub fn from_u128(v: u128) -> SessionId {
+        SessionId(Uuid::from_u128(v))
+    }
```

Comment:
Can we give these a big warning in the docs? Stating that they are only exposed for the purpose of writing custom serialisers, but no other code should rely on this as we are not promising the internal representation of session ids to stay compatible with this. Otherwise our previous discussion of "meh, we can change them later if we feel the need" is not really going to be true^^